### PR TITLE
Fkms rpi4

### PR DESCRIPTION
**Pull request by joolswills**
Here we go - time to re-integrate our dev branch

as I've been resyncing the branch with master I've had to resolve some conflicts here and there so it's worth giving this a review before we merge.

Thanks to everyone who has helped - including @psyke83 @cmitu @hhromic and everyone else

### CODE REVIEW

File: scriptmodules/system.sh [hunk 1]
```diff
@@ -101,8 +101,8 @@ function get_os_version() {
                 __platform_flags+=" xbian"
             fi
 
-            # we provide binaries for RPI on Raspbian 9/10
-            if isPlatform "rpi" && compareVersions "$__os_debian_ver" gt 8 && compareVersions "$__os_debian_ver" lt 11; then
+            # we provide binaries for RPI on Raspbian 9
+            if isPlatform "rpi" && compareVersions "$__os_debian_ver" gt 8 && compareVersions "$__os_debian_ver" lt 10; then
```
Comment (joolswills) [inline scriptmodules/system.sh]: I will likely re-enable binaries on master after we merge this for buster

File: scriptmodules/helpers.sh [hunk 2]
```diff
@@ -232,7 +232,7 @@ function getDepends() {
 
         if [[ "$md_mode" == "install" ]]; then
             # make sure we have our sdl1 / sdl2 installed
-            if ! isPlatform "x11" && [[ "$required" == "libsdl1.2-dev" ]] && hasPackage libsdl1.2-dev $(get_pkg_ver_sdl1) "ne"; then
+            if ! isPlatform "x11" && ! isPlatform "mesa" && [[ "$required" == "libsdl1.2-dev" ]] && hasPackage libsdl1.2-dev $(get_pkg_ver_sdl1) "ne"; then
```
[the remark this answers is not in the document]
Comment (joolswills) [reply scriptmodules/helpers.sh]: Yeah. I think this might need changing.

**Comment by joolswills** [comment]
Thanks

File: scriptmodules/emulators/reicast/reicast.sh [hunk 3]
```diff
@@ -79,11 +81,12 @@ if [[ ! -f "$biosdir/dc_boot.bin" ]]; then
...
+[[ -n "$XRES" ]] && params+=(-config x11:width=$XRES -config x11:height=$YRES)
 getAutoConf reicast_input && params+=($(mapInput))
 [[ -n "$AUDIO" ]] && params+=(-config audio:backend=$AUDIO -config audio:disable=0)
 [[ -n "$ROM" ]] && params+=(-config config:image="$ROM")
 if [[ "$AUDIO" == "oss" ]]; then
-    aoss "$rootdir/emulators/reicast/bin/reicast" "${params[@]}" >/dev/null
+    aoss "$rootdir/emulators/reicast/bin/reicast" "${params[@]}"
 else
-    "$rootdir/emulators/reicast/bin/reicast" "${params[@]}" >/dev/null
+    "$rootdir/emulators/reicast/bin/reicast" "${params[@]}"
```
[the remark this answers is not in the document]
Comment (psyke83) [reply scriptmodules/emulators/reicast/reicast.sh]: Are you sure? The sighandler patch should make this unnecessary. Removing the redirects allows us to have working logging for the application.

[the remark this answers is not in the document]
Comment (cmitu) [reply scriptmodules/emulators/reicast/reicast.sh]: Forgot this is swallowed by `runcommand`, I'll re-test.

[the remark this answers is not in the document]
Comment (cmitu) [reply scriptmodules/emulators/reicast/reicast.sh]: @psyke83 you're right, I've reverted the change.

**Comment by joolswills** [comment]
Yay!

**Pull request merged**